src/profiler/profiler.c

```diff
@@ -67,6 +67,43 @@ void xdebug_profiler_rinit(void)
 	XG_PROF(active) = 0;
 }
 
+static uint64_t profiler_nanotime_get()
+{
+	if (XG_PROF(profiler_nanotime_snapshot) == 0) {
+
+		// debug, if there is no snapshot, use the current time
+		return xdebug_get_nanotime() - XG_PROF(profiler_nanotime_penalty);
+
+		// emit an error here, code should never get here
+		fprintf(stderr, "Profiler nanotime snapshot was not created!");
```

Comment:
what is the best way to emit an error and stop profiler/php execution?

Reply:
Nothing should abort anything.